aws/src/integration/java/org/apache/iceberg/aws/lakeformation/LakeFormationTestBase.java

```diff
@@ -357,8 +359,20 @@ String getRandomTableName() {
     return LF_TEST_TABLE_PREFIX + UUID.randomUUID().toString().replace("-", "");
   }
 
-  private static void waitForIamConsistency() throws Exception {
-    Thread.sleep(IAM_PROPAGATION_DELAY); // sleep to make sure IAM up to date
+  private static void waitForIamConsistency(String roleName, String policyName) {
+    Awaitility.await()
+        .pollDelay(Duration.ofMillis(1000))
+        .atMost(Duration.ofSeconds(10))
+        .until(
+            () ->
+                iam.getRolePolicy(
+                        GetRolePolicyRequest.builder()
+                            .roleName(roleName)
+                            .policyName(policyName)
+                            .build())
+                    .roleName()
+                    .equalsIgnoreCase(roleName));
+    ; // wait to make sure IAM up to date
```

Comment:
that seeems like a weird formatting. Maybe just put the comment at the top in L363